src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -154,6 +155,12 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 							   "relation does not exist")));
 	}
 
+	/* We need to lock partitions too */
+	if (PartitionedTable(relationId))
```

Comment:
Ah, `master_create_distributed_table ` again :/ With 9.0, we should remove all of these all APIs..